app/controllers/api/v1/projects_controller.rb

```diff
@@ -24,10 +25,18 @@ def project_params
 
   def save_project
     populate_project_from_forge(project_params[:repo_url], true)
-    create_params(@project) if @project
+    assign_params(@project)
     @project = ProjectBuilder.new(current_user, @project_params || {}).fetch_project
   end
 
+  def assign_params(project)
+    return unless project
+
+    project.name = params[:name] if params[:name]
+    project.coverity_project_id = params[:coverity_project_id] if params[:coverity_project_id]
```

Comment:
We don't need the `if` conditions here. `project.name` will be nil `if params[:name]` is nil. Same goes for coverity_project_id. There is no value in checking for the presence of these 2 values.

Reply:
Move this code back into `save_project`.